process.rb

```diff
@@ -244,17 +251,29 @@ def slugify(word)
       total_contributions: referendum.calculation(:opposing_total) || [],
     ))
   end
+
+  supporting_total = referendum.calculation(:supporting_total) || 0
+  opposing_total = referendum.calculation(:opposing_total) || 0
+  ContributionsByOrigin[referendum.election_name][:race_totals] ||= []
+  ContributionsByOrigin[referendum.election_name][:race_totals].append({
+    'title': "Measure #{referendum['Measure_number']}",
+    'type': 'referendum',
+    'slug': title,
+    'amount': supporting_total + opposing_total
+  })
 end
 
 build_file('/_data/totals.json') do |f|
   f.puts JSON.pretty_generate(
     Hash[ELECTIONS.map do |election_name, election|
-      [
-        election_name,
-        ContributionsByOrigin.fetch(election_name, {}).merge(
-          largest_independent_expenditures: election.calculation(:largest_independent_expenditures)
-        )
-      ]
+      totals = ContributionsByOrigin.fetch(election_name, {})
+      totals[:largest_independent_expenditures] = election.calculation(:largest_independent_expenditures)
+
+      # Grab top 3 most expensive races
+      totals[:most_expensive_races] = totals[:race_totals].sort_by {|v| -v[:amount]}[0..2]
+      totals.delete(:race_totals)
```

Comment:
Ah, alas, it's maybe a bit of a smell that `ContributionsByOrigin` now just includes a bunch of global state not at all related to our `ContributionsByOrigin`. At some point maybe we should have calculators that can depend on each other to produce stuff like this.

Reply:
yea. You almost want a DAG of data jobs with varying levels of aggregations. election aggregations depend on the race-level aggregations.